docs/en/administration/Query_planning.md

```diff
@@ -237,6 +237,26 @@ REFRESH ASYNC
 AS SELECT /*+ SET_VAR(query_timeout=500) */ * from dual;
 ~~~
 
+### User-defined variable hint
+
+You can set one or more [user-defined variables](../reference/user_defined_variables.md) by using the `SET_USER_VARIABLE` hint in the form of the syntax `/*+ SET_USER_VARIABLE(...) */` in SELECT statements, or in the SELECT clause.
+
+If a query references the result of a scalar subquery or scalar expression, you can use a user-defined variable hint to set that subquery or expression as a user-defined variable and then run the query. This not only can help avoid duplicate computation, but also does not influence the entire session because user-defined variable hints take effect at the statement level, unlike the [general usage of user-defined variables](../reference/user_defined_variables.md) that take effect at the session level.
+
+#### Syntax
+
+~~~SQL
+[...] SELECT [/*+ SET_USER_VARIABLE(@var_name = expr [, @var_name = expr]*) */] ...
+~~~
+
+#### Examples
+
+The following query references scalar subqueries `select max(age) from users` and `select min(name) from users`, so you can use user-defined variable hints to set these two subqueries as user-defined variables, and then run the query.
```

Comment:
so you can use user-defined variable hints to -- so you can use the `SET_USER_VARIABLE` hint to